Gems/ROS2/Code/Source/Sensor/Events/PhysicsBasedSource.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <ROS2/Sensor/Events/PhysicsBasedSource.h>
+#include <ROS2/Sensor/SensorConfiguration.h>
+
+namespace ROS2
+{
+    void PhysicsBasedSource::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<PhysicsBasedSource>()->Version(1)->Field("Source enabled", &PhysicsBasedSource::m_sourceEnabled);
+
+            if (auto editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<PhysicsBasedSource>("Physics Based Source", "Sensor event source based on physics callback")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &PhysicsBasedSource::m_sourceEnabled,
+                        "Source enabled",
+                        "Enable/disable event source");
+            }
+        }
+    }
+
+    void PhysicsBasedSource::Activate()
+    {
+        m_onSceneSimulationEvent = AzPhysics::SceneEvents::OnSceneSimulationFinishHandler(
+            [this](AzPhysics::SceneHandle sceneHandle, float deltaTime)
+            {
+                if (!m_sourceEnabled)
+                {
+                    return;
+                }
+                m_sourceEvent.Signal(sceneHandle, deltaTime);
+            });
+
+        auto sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        AzPhysics::SceneHandle sceneHandle = sceneInterface->GetSceneHandle(AzPhysics::DefaultPhysicsSceneName);
+        sceneInterface->RegisterSceneSimulationFinishHandler(sceneHandle, m_onSceneSimulationEvent);
+    }
+
+    void PhysicsBasedSource::Deactivate()
+    {
+        m_onSceneSimulationEvent.Disconnect();
+    }
+
+    void PhysicsBasedSource::Configure(const SensorConfiguration& sensorConfiguration)
```

Comment:
I think that control logic of publishing should not be a part of the source.
m_publishingEnables whether the data is published, not whether it is acquired in the first place. Removing Configure seems reasonable in this EventSource hierarchy.

I believe that whether the source produces events should be controlled only by Active state of Component which uses it.